tapchannel/aux_invoice_manager_test.go

```diff
@@ -0,0 +1,571 @@
+// asset units, and may have an scid as routing hint.
+func genCustomRecords(t *rapid.T,
+	rfqID rfqmsg.ID) (lnwire.CustomRecords, uint64) {
+
+	assetUnits := rapid.Uint64Range(10_000, 100_000).Draw(t, "asset_units")
+
+	balance := []*rfqmsg.AssetBalance{
+		rfqmsg.NewAssetBalance(
+			assetID(rapid.Byte().Draw(t, "asset_id")), assetUnits,
+		),
+	}
+
+	htlc := genHtlc(t, balance, rfqID)
+
+	customRecords, err := lnwire.ParseCustomRecords(htlc.Bytes())
+	require.NoError(t, err)
+
+	return customRecords, assetUnits
+}
+
+// genHtlc generates an istance of rfqmsg.Htlc with the provided asset amounts
+// and rfqID.
+func genHtlc(t *rapid.T, balance []*rfqmsg.AssetBalance,
+	rfqID rfqmsg.ID) *rfqmsg.Htlc {
+
+	// Introduce a 1/5 chance of no rfqID in this htlc.
+	noRfqID := rapid.Uint16().Draw(t, "has_rfqid_probability")
+	if noRfqID%5 == 0 {
+		return rfqmsg.NewHtlc(balance, fn.None[rfqmsg.ID]())
+	}
+
+	// Introduce a 1/5 chance of a mismatch in the expected and actual htlc
+	// rfqID.
+	rfqIDMatch := rapid.Uint16().Draw(t, "rfqid_match_probability")
+	if rfqIDMatch%5 == 0 {
+		return rfqmsg.NewHtlc(
+			balance,
+			fn.Some(
+				dummyRfqID(
+					rapid.IntRange(0, 255).Draw(t, "scid"),
+				),
+			),
+		)
+	}
+
+	return rfqmsg.NewHtlc(balance, fn.Some(rfqID))
+}
+
+// genRequest generates an InvoiceHtlcModifyRequest with random values. This
+// method also returns the assetUnits and the rfqID used by the htlc.
+func genRequest(t *rapid.T) (lndclient.InvoiceHtlcModifyRequest, uint64,
+	rfqmsg.ID) {
+
+	request := lndclient.InvoiceHtlcModifyRequest{}
+
+	rfqID := genRandomRfqID(t)
+
+	request.Invoice = genInvoice(t)
+	wireRecords, assetUnits := genCustomRecords(t, rfqID)
+	request.WireCustomRecords = wireRecords
+	request.ExitHtlcAmt = lnwire.MilliSatoshi(request.Invoice.ValueMsat)
+
+	return request, assetUnits, rfqID
+}
+
+// genRequests generates a random array of requests to be processed by the
+// AuxInvoiceManager. It also returns the rfq map with the related rfq quotes.
+func genRequests(t *rapid.T) ([]lndclient.InvoiceHtlcModifyRequest,
+	rfq.BuyAcceptMap) {
+
+	rfqMap := rfq.BuyAcceptMap{}
+
+	len := rapid.IntRange(0, 5).Draw(t, "requestsLen")
+	requests := make([]lndclient.InvoiceHtlcModifyRequest, len)
+
+	for i := range requests {
+		req, numAssets, scid := genRequest(t)
+		requests[i] = req
+
+		genBuyQuotes(
+			t, rfqMap, numAssets, uint64(req.Invoice.ValueMsat),
+			scid,
+		)
+	}
+
+	return requests, rfqMap
+}
+
+// genBuyQuotes populates the provided map of rfq quotes with the desired values
+// for a specific
+func genBuyQuotes(t *rapid.T, rfqMap rfq.BuyAcceptMap, units, amtMsat uint64,
+	scid rfqmsg.ID) {
+
+	var peer route.Vertex
+	askPrice := lnwire.MilliSatoshi(0)
+
+	// Introduce a 1/8 chance that the quote's peerID is not correct.
+	noPeerMatch := rapid.Uint16().Draw(t, "nodeID_mismatch_probability")
+	if noPeerMatch%8 == 0 {
+		peer = route.Vertex{3, 1, 4}
```

Comment:
Would be good to further randomize the negative vertex values we're using as well. I think just generating 2 random bytes and converting that to a vertex would be sufficient.